x-pack/legacy/plugins/index_management/public/components/template_form/steps/step_logistics.tsx

```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useState, useEffect } from 'react';
+import {
+  EuiComboBox,
+  EuiComboBoxOptionProps,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTitle,
+  EuiButtonEmpty,
+  EuiSpacer,
+  EuiDescribedFormGroup,
+  EuiFormRow,
+  EuiFieldText,
+  EuiText,
+  EuiFieldNumber,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { get } from 'lodash';
+import { Template } from '../../../../common/types';
+import { INVALID_CHARACTERS } from '../../../../common/constants';
+import { templatesDocumentationLink } from '../../../lib/documentation_links';
+import { loadIndexPatterns } from '../../../services/api';
+import { StepProps } from '../types';
+
+const indexPatternIllegalCharacters = INVALID_CHARACTERS.join(' ');
+
+export const StepLogistics: React.FunctionComponent<StepProps> = ({
+  template,
+  updateTemplate,
+  errors,
+  isEditing,
+}) => {
+  const { name, order, version, indexPatterns } = template;
+  const { name: nameError, indexPatterns: indexPatternsError } = errors;
+
+  const [allIndexPatterns, setAllIndexPatterns] = useState<Template['indexPatterns']>([]);
+
+  const getIndexPatterns = async () => {
+    const indexPatternObjects = await loadIndexPatterns();
```

Comment:
@sebelga can you point in the right direction in the CCR code to take a look at this? The current implementation is the approach watcher uses to load index patterns when creating a threshold alert.

Reply:
So the question is, you seem to be loading the Kibana index patterns. And if I am not wrong the patterns ES is expecting are different.

In CCR they are called "LeaderIndexPatterns"

https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/cross_cluster_replication/public/app/components/auto_follow_pattern_form.js#L121

Reply:
@sebelga I understand, thanks.

It looks like CCR only allows custom values and then has an info callout when entered, showing potentially what kind of indices the pattern would match. What do you think if we follow the same pattern here?

<img width="623" alt="Screen Shot 2019-08-21 at 8 43 29 AM" src="https://user-images.githubusercontent.com/5226211/63432849-fa924200-c3ef-11e9-848f-2ee75ccade27.png">

Reply:
Yes, it only allows custom values because patterns are custom by default 😊 (there is nowhere a user can save patterns in ES).

I am not sure that we need to add the preview, its purpose was mainly for the "prefix" and "suffix" that could be added.

Reply:
Right, makes sense.

Fair enough :) Thanks for talking through this with me!

Reply:
no prob!